File: app/lib/models/pharmacy.dart
// lib/models/pharmacy.dart
import 'abonnement.dart';

class Pharmacy {
  final String id;
  final String nom;
  final String telephone;
  final String? email;
  final String adresse;
  final String region;
  final String ville;
  final Abonnement abonnement;
  final String lienGoogleMaps;
  final bool actif;

  Pharmacy({
    required this.id,
    required this.nom,
    required this.telephone,
    this.email,
    required this.adresse,
    required this.region,
    required this.ville,
    required this.abonnement,
    required this.lienGoogleMaps,
    required this.actif,
  });

  factory Pharmacy.fromJson(Map<String, dynamic> json) {
    return Pharmacy(
      id: json['id'],
      nom: json['nom'],
      telephone: json['telephone'],
      email: json['email'],
      adresse: json['adresse'],
      region: json['region'],
      ville: json['ville'],
      abonnement: Abonnement.fromJson(json['abonnement']),
      lienGoogleMaps: json['lien_google_maps'],
      actif: json['actif'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
      'telephone': telephone,
      'email': email,
      'adresse': adresse,
      'region': region,
      'ville': ville,
      'abonnement': abonnement.toJson(),
      'lien_google_maps': lienGoogleMaps,
      'actif': actif,
    };
  }
}

File: backend/models/Pharmacy.js
const mongoose = require('mongoose');

const noteSchema = new mongoose.Schema({
  moyenne: { type: Number, default: 0 },
  count: { type: Number, default: 0 }
});

const abonnementSchema = new mongoose.Schema({
  plan: { type: String, enum: ["1 mois", "3 mois", "6 mois", "12 mois"], required: true },
  date_debut: { type: Date, required: true },
  date_fin: { type: Date, required: true },
  actif: { type: Boolean, default: true }
});

const pharmacySchema = new mongoose.Schema({
  nom: { type: String, required: true },
  region: { type: mongoose.Schema.Types.ObjectId, ref: 'Region', required: true },
  ville: { type: mongoose.Schema.Types.ObjectId, ref: 'City', required: true },
  adresse: { type: String, required: true },
  telephone: { type: String, required: true, unique: true },
  email: { type: String, required: true, unique: true },
  mot_de_passe: { type: String, required: true }, // Haché avec bcrypt
  abonnement: { type: abonnementSchema, required: true },
  lien_google_maps: { type: String, required: true },
  note: { type: noteSchema, default: () => ({}) },
  fcm_token: { type: String, default: null },
  actif: { type: Boolean, default: true },
  date_inscription: { type: Date, default: Date.now },
  date_derniere_connexion: { type: Date }
});

module.exports = mongoose.model('Pharmacy', pharmacySchema);
